### PR TITLE
feat: base cypress structure

### PR DESCRIPTION
**Pull request by EkaterinaSokolova-V**

### CODE REVIEW

File: mw-webapp/cypress/fixtures/allWaysFixture.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "endpoint": "/ways",
+    "waysTitle": "Ways"
```
Comment (tritonJS826) [inline mw-webapp/cypress/fixtures/allWaysFixture.json:3]: I think it will be better to reorganize fixtures in future, but I think it is good enough for now

File: mw-webapp/cypress/scopesSelectors/headerSelectors.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { headerAccessIds } from "cypress/accessIds/headerAccessIds";
+import { getDataCy } from "src/utils/cyTesting/getDataCy";
+
+export const headerSelectors = {
+    items: {
+        getLogoBtn: () => cy.get(getDataCy(headerAccessIds.items.allWaysLogoBtn)),
```
Comment (tritonJS826) [inline mw-webapp/cypress/scopesSelectors/headerSelectors.ts:6]: I think there it will be nice to have  direct connection between id an selector, we can achieve it by just changing prefixes: 
for example:
id:  allWaysLogoBtn
selector: getAllWaysLogoBtn


i think it will be easier to work with such system in the future, what do you think?

File: mw-webapp/src/component/header/Header.tsx [hunk 3]
```diff
@@ -147,6 +148,7 @@ export const Header = (props: HeaderProps) => {
       <Link
         className={styles.logo}
         path={pages.allWays.getPath({})}
+        dataCy={headerAccessIds.items.allWaysLogoBtn}
```
Comment (tritonJS826) [inline mw-webapp/src/component/header/Header.tsx:151]: Also logo in the header does it is not a button, it is a link, but it does not matter to be honest, so: 
`headerAccessIds.items.allWaysLogoBtn` -> `headerAccessIds.items.logo`

File: mw-webapp/src/logic/aboutProjectPage/TeamMember/teamMembers.ts [hunk 4]
```diff
@@ -6,7 +6,7 @@ export const teamMembers: TeamMemberType[] = [
     id: 1,
     name: "Viktar Veratsennikau",
     description: "Founder",
-    imageUrl: "https://lh3.googleusercontent.com/drive-viewer/AKGpihawXxK1rm9vcGMcu2iUZ5RtxoIPAwgOvNuAtoOGcZKfoJ6O7yAaUTZ8k5sDwTNOxg4Lgcwch_nqlroh9t5uV94bNzIHjyjSsw=s1600-v0",
+    imageUrl: "",
```
Comment (tritonJS826) [inline mw-webapp/src/logic/aboutProjectPage/TeamMember/teamMembers.ts:9]: why did you remove my image? Don't you like it :)

Comment (tritonJS826) [reply mw-webapp/src/logic/aboutProjectPage/TeamMember/teamMembers.ts:9]: try to replace it on the `https://lh3.google.com/u/0/d/18oHI9KoiaYvd_UowHyqsJbDLLhmuxPxr=w2000-h2284-iv2` if it does not work for now

File: mw-webapp/src/logic/allWaysPage/AllWaysPage.tsx [hunk 5]
```diff
@@ -195,6 +196,7 @@ export const AllWaysPage = () => {
         <Title
           level={HeadingLevel.h2}
           text={`${LanguageService.allWays.waysTable.leftTitle[language]} (${allWays.length})`}
+          dataCy={allWaysAccessIds.titles.allWays}
```
Comment (tritonJS826) [inline mw-webapp/src/logic/allWaysPage/AllWaysPage.tsx:199]: waht do you think about a little bit shorter syntax `allWaysAccessIds` -> `allWaysId`

(we dont need to use 's' because this structure is not a list)

File: mw-webapp/pnpm-lock.yaml
Comment (Ekaterina1994) [inline mw-webapp/pnpm-lock.yaml:1]: Why this file was changed?

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]
Need resolve all conflicts

File: mw-webapp/cypress/fixtures/allWaysFixture.json (hunk 1 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/fixtures/allWaysFixture.json:3]: Ok, let's leave it like that for now.

File: mw-webapp/pnpm-lock.yaml
Comment (EkaterinaSokolova-V) [reply mw-webapp/pnpm-lock.yaml:1]: Good question) I have no idea. I will replace it with the original file.

File: mw-webapp/src/logic/aboutProjectPage/TeamMember/teamMembers.ts (hunk 4 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/src/logic/aboutProjectPage/TeamMember/teamMembers.ts:9]: I got an error about too many characters, the only way to avoid it was to delete this line. 
Replaced.

File: mw-webapp/src/logic/allWaysPage/AllWaysPage.tsx (hunk 5 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/src/logic/allWaysPage/AllWaysPage.tsx:199]: Good point

File: mw-webapp/src/component/header/Header.tsx (hunk 3 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/src/component/header/Header.tsx:151]: If I changed ID to 'logo', then in this case I also leave only 'logo' for the selector, right?
Id: logo
Selector: getLogo

Comment (tritonJS826) [reply mw-webapp/src/component/header/Header.tsx:144]: not clear what does this lines mean 
```
Id: logo
Selector: getLogo
```

Comment (tritonJS826) [reply mw-webapp/src/component/header/Header.tsx:144]: yeah, good

File: mw-webapp/src/logic/aboutProjectPage/TeamMember/teamMembers.ts (hunk 4 above)
Comment (tritonJS826) [reply mw-webapp/src/logic/aboutProjectPage/TeamMember/teamMembers.ts]: Do you think it is a right solution?

**Comment by EkaterinaSokolova-V** [comment]
@tritonJS826  @Ekaterina1994 please have a look

**Review by Ekaterina1994 — APPROVED** [review]

File: mw-webapp/cypress/accessIds/allWaysAccessIds.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export const allWaysAccessIds = {
+    titles: {
+        allWays : 'ways-title',
```
Comment (Ekaterina1994) [inline mw-webapp/cypress/accessIds/allWaysAccessIds.ts:3]: Is it possible to use camelCase instead of kebab-case for consistancy. If yes, please  check everywhere

File: mw-webapp/cypress/accessIds/allWaysAccessIds.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export const allWaysAccessIds = {
+    titles: {
```
Comment (Ekaterina1994) [inline mw-webapp/cypress/accessIds/allWaysAccessIds.ts:2]: let's group data by skopes (block) on the page. For example,  title could be inside AllWaysTable block, and other title could be inside FilterBlock , etc. The same on other pages )

File: mw-webapp/cypress/e2e/allWays.cy.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
+import {allWaysSelectors} from "cypress/scopesSelectors/allWaysSelectors";
+import pageData from "../fixtures/allWaysFixture.json"
```
Comment (Ekaterina1994) [inline mw-webapp/cypress/e2e/allWays.cy.ts:2]: please use absolute path if its possible

File: mw-webapp/cypress/e2e/allWays.cy.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+import {headerSelectors} from "cypress/scopesSelectors/headerSelectors";
+
+describe.skip('All Ways scope tests', () => {
+
+  beforeEach(() => {
+    cy.visit('/');
+  });
+
+  it('NoAuth_AllWays_Open', () => {
```
Comment (Ekaterina1994) [inline mw-webapp/cypress/e2e/allWays.cy.ts:11]: please don't use snake_case in describing tests you can use spaces between words

File: mw-webapp/cypress/e2e/allWays.cy.ts [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    headerSelectors.items.getLogoBtn().click();
+
+    cy.url().should('include', pageData.endpoint);
+    allWaysSelectors.titles.getAllWaysTitle().should('contain', pageData.waysTitle);
```
Comment (Ekaterina1994) [inline mw-webapp/cypress/e2e/allWays.cy.ts:15]: if you will restructure accsessIds then it will be more clear to use it. For example here => allWaysSelectors.AllWaysTable.title.should("contain"...

File: mw-webapp/cypress/e2e/home.cy.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+
+describe('Home page scope tests', () => {
+
+  beforeEach(() => {
+    cy.visit('/');
+  });
+
+  it('homePage_Open', () => {
+    homeSelectors.pageTitle.getTitle().should('contain', pageData.title);
+    homeSelectors.items.getStartBtn().should('have.text', pageData.btnTitle);
```
Comment (Ekaterina1994) [inline mw-webapp/cypress/e2e/home.cy.ts:12]: PLease don't use short words (btn, lvl etc). Its not clear what they mean. Its just few symbols but code will be more clear

File: mw-webapp/cypress/fixtures/homeFixture.json [hunk 12]
```diff
@@ -0,0 +1,4 @@
+{
+    "title": "Accelerate your goals with an asynchronous learning app",
+    "btnTitle": "Start for free!"
```
Comment (Ekaterina1994) [inline mw-webapp/cypress/fixtures/homeFixture.json:3]: I don't know bur what will be if developers will change button value on frontend from "Start for free!" to "Something other". Is it will be okey or will be needed to update these texts too?  Is it possible to use our dictionaries .json?

File: mw-webapp/src/component/header/Header.tsx [hunk 13]
```diff
@@ -162,6 +163,7 @@ export const Header = (props: HeaderProps) => {
       <Link
         className={styles.logo}
         path={pages.home.getPath({})}
+        dataCy={headerAccessIds.items.allWaysLogoBtn}
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/header/Header.tsx:166]: it's little bit strange that link is called logoBtn

File: mw-webapp/cypress/accessIds/allWaysAccessIds.ts (hunk 6 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/accessIds/allWaysAccessIds.ts:3]: Ok

File: mw-webapp/cypress/e2e/allWays.cy.ts (hunk 8 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/allWays.cy.ts:2]: done

File: mw-webapp/cypress/e2e/allWays.cy.ts (hunk 9 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/allWays.cy.ts:11]: Как я поняла, тут должен быть ID теста из нашей документации, они у нас в таком виде сделаны. Но можно то же самое и camel case. Как сделать?

File: mw-webapp/cypress/e2e/home.cy.ts (hunk 11 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/home.cy.ts:12]: Ok.

File: mw-webapp/cypress/fixtures/homeFixture.json (hunk 12 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/fixtures/homeFixture.json:3]: Изучу этот вопрос и исправлю.

File: mw-webapp/src/component/header/Header.tsx (hunk 13 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/src/component/header/Header.tsx]: Changed.

File: mw-webapp/cypress/accessIds/allWaysAccessIds.ts (hunk 7 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/accessIds/allWaysAccessIds.ts:2]: Не понимаю, какой должна быть структура. Вот такая:
AllWaysTable: {         
        allWays : 'waysTitle',
        totalFound : 'totalFoundTitle'
    }

Filter: {         
    .....
    }
};

File: mw-webapp/cypress/e2e/allWays.cy.ts (hunk 10 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/allWays.cy.ts:15]: И для allWaysSelectors:
export const allWaysSelectors = {
   allWaysTable: {
        getTitle: () => cy.get(getDataCy(allWaysAccessIds.allWaysTable.title)),
    }

};

allWaysSelectors.AllWaysTable.getTitle().should("contain"...

Comment (Ekaterina1994) [reply mw-webapp/cypress/e2e/allWays.cy.ts:15]: Да,)

File: mw-webapp/cypress/e2e/allWays.cy.ts (hunk 9 above)
Comment (Ekaterina1994) [reply mw-webapp/cypress/e2e/allWays.cy.ts:11]: @tritonJS826  Как лучше?

File: mw-webapp/cypress/accessIds/allWaysAccessIds.ts (hunk 7 above)
Comment (Ekaterina1994) [reply mw-webapp/cypress/accessIds/allWaysAccessIds.ts:2]: Discuused in personal messages

File: mw-webapp/cypress/e2e/allWays.cy.ts (hunk 10 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/e2e/allWays.cy.ts]: done

File: mw-webapp/cypress/fixtures/homeFixture.json (hunk 12 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/fixtures/homeFixture.json]: done

File: mw-webapp/cypress/accessIds/allWaysAccessIds.ts [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+    filterView: {
+        statusSelect: 'statusSelect',
+        cardViewButton: 'cardViewButton',
+        tableViewButton: 'tableViewButton'
+    },
+
+    ways: {
+        
+    }
+};
```
Comment (Ekaterina1994) [inline mw-webapp/cypress/accessIds/allWaysAccessIds.ts:16]:
```suggestion
export const allWaysAccessIds = {
allWaysTable: {
  title: ...,
  totalAmountTitle: ...
},
  filterViewBlock: {
        filterByStatus: ...,
        cardViewButton: ...',
        tableViewButton: ...
    },

   };
```

File: mw-webapp/cypress/accessIds/headerAccessIds.ts [hunk 15]
```diff
@@ -0,0 +1,10 @@
+export const headerAccessIds = {
+    link: {
+        logo: 'homeLogo',
+    },
+
+    buttons: {
+        themeSwitcher: 'themeSwitcher' 
+    }
+    
+};
```
Comment (Ekaterina1994) [inline mw-webapp/cypress/accessIds/headerAccessIds.ts:10]:
```suggestion
export const headerAccessIds = {
      logo: 'homeLogo',
      burgerMenu: ...,
      avatar: ....,
      settings: {
        themeSwitcher: ...,
        .....
      }
    
};
```

File: mw-webapp/cypress/accessIds/allWaysAccessIds.ts (hunk 14 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/accessIds/allWaysAccessIds.ts]: @Ekaterina1994 done

File: mw-webapp/cypress/accessIds/headerAccessIds.ts (hunk 15 above)
Comment (EkaterinaSokolova-V) [reply mw-webapp/cypress/accessIds/headerAccessIds.ts]: done

**Review by tritonJS826 — APPROVED** [review]
